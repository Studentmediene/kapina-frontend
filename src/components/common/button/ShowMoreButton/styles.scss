.showMoreButton {
  font-size: 24px;
  padding: 14px 0;
  border-width: 0;
  cursor: pointer;
}

.arrow {
  width: 20px;
  transition: 100ms linear all;
}

.rotated {
  transform: rotate(180deg);
}
