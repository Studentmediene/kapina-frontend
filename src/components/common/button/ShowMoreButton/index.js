import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './styles.scss';
import arrowImage from './arrow-down.svg';

const ShowMoreButton = props => (
  <button className={styles.showMoreButton} onClick={props.onClick}>
    {`${props.text} `}
    <img
      src={arrowImage}
      alt="Arrow"
      className={classNames(styles.arrow, {
        [styles.rotated]: props.active,
      })}
    />
  </button>
);

ShowMoreButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool,
  text: PropTypes.string.isRequired,
};

export default ShowMoreButton;
