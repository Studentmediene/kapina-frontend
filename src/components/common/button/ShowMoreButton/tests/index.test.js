import React from 'react';
import { shallow } from 'enzyme';

import ShowMoreButton from '../';

describe('<ShowMoreButton />', () => {
  it('renders correctly when active', () => {
    const mockProps = {
      active: true,
      onClick: jest.fn(),
      text: 'text',
    };
    const tree = shallow(<ShowMoreButton {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when inactive', () => {
    const mockProps = {
      active: false,
      onClick: jest.fn(),
      text: 'text',
    };
    const tree = shallow(<ShowMoreButton {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });
});
