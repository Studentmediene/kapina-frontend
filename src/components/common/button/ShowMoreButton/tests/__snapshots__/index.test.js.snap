// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowMoreButton /> renders correctly when active 1`] = `
<button
  className="showMoreButton"
  onClick={[MockFunction]}
>
  text 
  <img
    alt="Arrow"
    className="arrow rotated"
    src="arrow-down.svg"
  />
</button>
`;

exports[`<ShowMoreButton /> renders correctly when inactive 1`] = `
<button
  className="showMoreButton"
  onClick={[MockFunction]}
>
  text 
  <img
    alt="Arrow"
    className="arrow"
    src="arrow-down.svg"
  />
</button>
`;
